core/rawdb/accessors_chain.go

```diff
@@ -708,10 +708,12 @@ func (r *receiptLogs) DecodeRLP(s *rlp.Stream) error {
 // DeriveLogFields fills the logs in receiptLogs with information such as block number, txhash, etc.
 func deriveLogFields(receipts []*receiptLogs, hash common.Hash, number uint64, txs types.Transactions) error {
 	logIndex := uint(0)
-	if len(txs) != len(receipts) {
+	// The receipts may include an additional "block finalization" receipt (only IBFT)
+	if !(len(txs) == len(receipts) || len(txs)+1 == len(receipts)) {
```

Comment:
Should we verify that if len(receipts) == len(txs) + 1 then the block finalization receipt is exactly the last one from the array?